### PR TITLE
Fix measure_all_oneshot (Qubit(0)) that yields |01>

### LINKED ISSUE #27092: measure_all_oneshot of Qubit(0) yields |01>
If I `measure_all_oneshot` a Qubit like |0>, |1>, or a superposition of the two
I get
`|01>`
or
`|11>`

Here's a minimal example
```
from sympy.physics.quantum.qubit import Qubit, measure_all, measure_all_oneshot
q = Qubit(0)
measure_all_oneshot(q)
|01>
```
while if I do a `measure_all`
```
measure_all(q)
[(|0>, 1)]
```
if I change the code and add `nqubits` on line quibit.py line 807 
like so 
`return Qubit(IntQubit(result, nqubits=int(math.log(max(m.shape), 2) + .1))) `

I get 
`|0>`
or
`|1>`

If this fix is reasonable I can push the code in a PR :)

### PR DESCRIPTION
**Pull request by groncarolo**
Fixes #27092

Fix measure_all_oneshot (Qubit(0)) that yields |01>
   
Fix the following case:
```
q = Qubit(0)
measure_all_oneshot(q)
|01>
 ```
Add also the tests for this use case

* physics.quantum
  * A bug in Qubit.measure_all_oneshot was fixed. Previously measuring `measure_all_oneshot(Qubit(0))` would return incorrectly `|01>`. Now `Qubit.measure_all_oneshot(Qubit(0))` returns correct results `|0>`

### CODE REVIEW

File: .mailmap [hunk 1]
```diff
@@ -658,6 +658,7 @@ Grace Su <grace.duansu@gmail.com>
 Gregory Ashton <gash789@gmail.com> <ga7g08@soton.ac.uk>
 Gregory Ksionda <ksiondag846@gmail.com>
 Grzegorz Świrski <sognat@gmail.com>
+Guido Roncarolo <guido.roncarolo@gmail.com>
```
Comment (oscarbenjamin) [inline .mailmap:661]: It should be like this:
```
Guido Roncarolo <guido.roncarolo@gmail.com> groncarolo <53181812+groncarolo@users.noreply.github.com>
```

Comment (groncarolo) [reply .mailmap:661]: Thanks!

**Comment by oscarbenjamin** [comment]
@hkve can you review this?

**Review by hkve — COMMENTED** [review]
Looks good! Remember to add [NO ENTRY](https://github.com/sympy/sympy/wiki/Writing-Release-Notes) to the release notes entry.

**Comment by oscarbenjamin** [comment]
Thanks. I think if this fixes a bug then it is better to add a release note rather than NO ENTRY. The release note needs to be formatted as explained by sympy-bot though: https://github.com/sympy/sympy/pull/27105#issuecomment-2366841507

**Comment by groncarolo** [comment]
Sorry for the delay, I had busier than usual days in the office

**Comment by oscarbenjamin** [comment]
No problem. The release note says:

* physics.quantum
  * Fix 27092
  * Add test for measure_all_oneshot

We don't need to mention adding a test. The note about fixing the issue should instead explain what is actually fixed. I don't quite understand the change well enough to write this myself but anyone who does can suggest the wording here. It would be something like:

* physics.quantum
  * A bug in `Qubit.measure_all_oneshot` was fixed. Previously if `<something>` then an incorrect result such as `<something>` would be returned. Now `Qubit.measure_all_oneshot` returns correct results for all `<something>`.

The release note will automatically have a link back to the PR which someone can use to find more information and go read the issue that is linked in the OP. Note that I edited the OP so that `27092` is replaced with `#27092` which appears in GitHub as #27092 (i.e. a hyperlink to the issue). That also mean that the GitHub recognises the "Fixes" line so that when this is merged the issue will be closed automatically.

**Comment by oscarbenjamin** [comment]
You can see examples of release notes here:

https://github.com/sympy/sympy/wiki/Release-Notes-for-1.13.0

(Not that all of them are particularly good.)

**Comment by groncarolo** [comment]
I hope this can do, thanks for the support!

**Comment by oscarbenjamin** [comment]
Looks good. Thanks @groncarolo and @hkve

**Pull request merged**